save-backend/src/main/kotlin/org/cqfn/save/backend/repository/TestExecutionRepository.kt

```diff
@@ -34,4 +35,14 @@ interface TestExecutionRepository : BaseEntityRepository<TestExecution> {
      * @return a list of [TestExecutionDto]s
      */
     fun findByExecutionId(executionId: Long, pageable: Pageable): List<TestExecution>
+
+    /**
+     * Returns a TestExecution matched by a set of fields
+     *
+     * @param executionId if of execution
+     * @param agentId id of an agent (numeric, as in the DB)
+     * @param filePath path to test file
+     * @return Optional TestExecution
+     */
+    fun findByExecutionIdAndAgentIdAndTestFilePath(executionId: Long, agentId: Long, filePath: String): Optional<TestExecution>
```

Comment:
I'd say no, because `executionId` is fixed for a snapshot of resources (e.g. specific git commit), and you can't have different files with same path. If file content changes over time with commits, it's hash changes, and we store a new test, but test execution for a particular execution should be unique

Reply:
Actually, we should take plugin_name into account here too, it will be added in #176